lib/plugins/azurepush.js

```diff
@@ -0,0 +1,323 @@
+        if (docs.length > 0) {
+          ctx.store.collection(env.azurepush_collection).update({_id: entry._id}, {$set: {settings: entry.settings}}, function updateAzurepush(err, doc) {
+            // *****REMOVE*****: Beta logging
+            if (err) {
+              logTrelloIssue(env, '0x05-Failed to update existing registration in mongodb');
+            }
+            // END: *****REMOVE*****
+
+            // TODO: Handle error
+            console.log(err, doc);
+          });
+        } else {
+          ctx.store.collection(env.azurepush_collection).insert(entry, function storeAzurepush(err, doc) {
+            // *****REMOVE*****: Beta logging
+            if (err) {
+              logTrelloIssue(env, '0x06-Failed to add new registration in mongodb');
+            }
+            // END: *****REMOVE*****
+            
+            // TODO: Handle error
+            console.log(err, doc);
+          });
+        }
+      });
+
+      // Verify/create index
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.announcement', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x07-Failed to verify index on anncouncement in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.alert', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x08-Failed to verify index on alert in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.info', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x09-Failed to verify index on info in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      
+      // More info on request body: https://msdn.microsoft.com/en-us/library/mt621153.aspx
+      var installation = {
+        installationId: request.installationId
+        , pushChannel: request.deviceToken
+        , templates: {
+          nightscout: {
+            tags: tags
+          }
+        }
+      };
+
+      // 'platform': https://developer.xamarin.com/guides/xamarin-forms/platform-features/device/#Device.OS
+      switch (request.platform) {
+        case 'iOS':
+          installation.platform = 'apns';
+          installation.templates.nightscout.body = "{\"aps\":{\"alert\":\"$(message)\",\"sound\":\"$(sound)\",\"category\":\"$(category)\"},\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"title\":\"$(title)\"}";
```

Comment:
if you use single quotes you could avoid all the escaping, maybe better to build as a json object and then convert to string?